src/tokenizer.rs

```diff
@@ -347,11 +347,39 @@ impl<'a> Tokenizer<'a> {
                 }
                 // numbers
                 '0'..='9' => {
-                    // TODO: https://jakewheat.github.io/sql-overview/sql-2011-foundation-grammar.html#unsigned-numeric-literal
-                    let s = peeking_take_while(chars, |ch| match ch {
-                        '0'..='9' | '.' => true,
+                    let mut seen_decimal = false;
+                    let mut s = peeking_take_while(chars, |ch| match ch {
+                        '0'..='9' => true,
+                        '.' => {
```

Comment:
Definitely don't go back through CI for this, but using pattern guards here will avoid some nesting:

```rust
let mut s = peeking_take_while(chars, |ch| match ch {
    '0'..='9' => true,
    '.' if !seen_decimal => { seen_decimal = true; true }
    '.' => false,
    _ => false,
});
```